platform/spire/src/setup.py

```diff
@@ -91,6 +94,37 @@ def admit_keyserver(ops: command.Operations) -> None:
         ssh_cmd(ops, "enable auth-monitor daemon on @HOST", node, "systemctl", "enable", "auth-monitor")
         ssh_cmd(ops, "start auth-monitor daemon on @HOST", node, "systemctl", "restart", "auth-monitor")
 
+@command.wrapop
+def redeploy_keyserver(ops: command.Operations) -> None:
+    "redeploy configuration for keyserver; restart keyserver"
+    config = configuration.get_config()
+    for node in config.nodes:
+        if node.kind != "supervisor":
+            continue
+        # delete the existing configs
+        ssh_rm(ops, "delete existing cluster config from @HOST", node, STATICS_DIR + "/cluster.conf")
+        ssh_rm(ops, "delete existing keyserver config from @HOST", node, CONFIG_DIR + "/setup.yaml")
+        # redeploy new config
+        ssh_upload_bytes(ops, "reupload cluster config to @HOST", node,
+            configuration.get_cluster_conf().encode(), STATICS_DIR + "/cluster.conf")
+        ssh_upload_path(ops, "reupload cluster setup to @HOST", node,
+                            configuration.Config.get_setup_path(), CONFIG_DIR + "/setup.yaml")
+        # restart the keyserver
+        ssh_cmd(ops, "restart keyserver on @HOST", node, "systemctl", "restart", "keyserver.service")
+
+@command.wrapop
+def redeploy_keyclients(ops: command.Operations) -> None:
+    "invalidate local configuration for keyclients; restart keyclients"
+    config = configuration.get_config()
+    for node in config.nodes:
+        # we do not delete the config on the supervisor because the supervisor
+        # client config and keyserver config are in the same location
+        if node.kind != "supervisor":
+            ssh_rm(ops, "delete existing cluster config from @HOST", node, CONFIG_DIR + "/cluster.conf")
+        ssh_rm(ops, "delete existing local config from @HOST", node, CONFIG_DIR + "/local.conf")
+        # restart local keyclient (will regenerate configs on restart)
+        ssh_cmd(ops, "restart keyclient daemon on @HOST", node, "systemctl", "restart", "keyclient.service")
+
```

Comment:
Do we want to ability to redeploy these individually?  If so then they need to be placed into the subcommand hierarchy somewhere

Reply:
To clarify, you mean to say that we may want to allow redeploy keyclients and redeploy keyserver as the subcommands and not redeploy each server individually?

If so I think it would be fine to support the sub commands, although, I'm not sure how often one may want to redeploy the config to the keyserver without doing so also to the keyclients.

Reply:
I was just wondering if you possibly meant to add subcommands for these functions and forgot; I have no opinion on it otherwise.

Reply:
Okay, I didn't mean to add them as sub-commands, and I don't think that separating the redeploy step into two phases will be necessary. I have decided to keep them under one command, but thanks for letting me know!